### PR TITLE
Carousel data complete

### PR DESCRIPTION
**Pull request by amardesi**

### CODE REVIEW

File: app/dbCache.rb [hunk 1]
```diff
@@ -99,6 +100,57 @@ def getViewsPerCampus
...
+# Unit Carousel: Compile item and view counts for all configured units. Campus may configure up to two (2) units
+# {"uclalaw"=> {item_count: 3408, view_count: 30000}, "cpcc"=>...}
+def getUnitCarouselStats
+  unitsInCars = $activeCampuses.map do |id, c|
+    unit = Unit[id]
+    attrs = JSON.parse(unit.attrs)
+    x = unitsFromContentCarConfig(JSON.parse(unit.attrs))
+    next x
+  end
+  unitsInCars.reject!{ |u| u.empty? }.flatten!.uniq!
```
Comment (martinhaye) [inline app/dbCache.rb]: So fancy.

File: app/dbCache.rb [hunk 2]
```diff
@@ -99,6 +100,57 @@ def getViewsPerCampus
...
+  #  map{|y| y.values}
+  # itemCounts = Hash[itemsInUnitCar.map(&:values).map(&:flatten)]
+  itemsInUnitCar = UnitCount.select_group(:unit_id).where(:unit_id=>unitsInCars).select_append{sum(:items_posted).as(item_count)}.map{|y| y.values}
+  viewsInUnitCar = UnitCount.select_group(:unit_id).where(:unit_id=>unitsInCars).select_append{sum(:hits).as(view_count)}.map{|y| y.values}
+  x = (itemsInUnitCar + viewsInUnitCar).group_by{|h| h[:unit_id]}.map{|k, v| v.reduce(Hash.new, :merge) }
+  return x.map{|h| [h[:unit_id], {:item_count=>h[:item_count], :view_count=>h[:view_count]}]}.to_h
+end
+
+def sumArrayCounts(array)
+  return array.inject(0) {|sum, h| sum + h[:count]}
```
Comment (martinhaye) [inline app/dbCache.rb]: Someday I have to understand this inject thing. I've seen it done, I just never stopped to figure it out :)

File: app/jsx/components/JournalCarouselComp.jsx [hunk 3]
```diff
@@ -33,6 +35,7 @@ class JournalCarouselComp extends React.Component {
   }
   render() {
     let p = this.props
+    let pluralItems = (p.item_count == 1) ? '' : 's'
```
Comment (martinhaye) [inline app/jsx/components/JournalCarouselComp.jsx]: Much nicer than "item(s)"

File: app/dbCache.rb (hunk 2 above)
Comment (amardesi) [reply app/dbCache.rb]: It should be renamed to accumulate. This is what I bookmarked a long time ago and has been very helpful: http://queirozf.com/entries/ruby-map-each-collect-inject-reject-select-quick-reference

**Pull request merged**